### PR TITLE
rename and update package dependencies, run formatter

### PR DESCRIPTION
**Pull request by mabiede**
As the pipeline fails and I cannot edit the PR #53 (thanks @mbacarella), here is an updated version:

- related to https://github.com/ocaml/opam-repository/pull/23311
- update `dune-project`
  - add missing dependency tls-lwt
  - update sendmail and colombe version due >= 0.7.0 (`LOGIN` mechanism)
- update lock file
- add `(implicit_transitive_deps false)` and specify dependencies

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -1,42 +1,71 @@
 (lang dune 2.7)
+
```
Comment (aronerben) [inline dune-project:2]: I'm guessing this is added by the dune-project formatter?

**Review by aronerben — APPROVED** [review]

**Comment by aronerben** [comment]
@joseferben @mikonieminen FYI, I'm merging this

File: dune-project (hunk 1 above)
Comment (mabiede) [reply dune-project:2]: yes, correct.

**Comment by dinosaure** [comment]
I would like to mention that a release of `mrmime` with some breaks will be done soon. You probably shoud integrate these breaks before a release 👍 . I can do something when the `mrmime` release is done.

**Comment by aronerben** [comment]
> I would like to mention that a release of `mrmime` with some breaks will be done soon. You probably shoud integrate these breaks before a release +1 . I can do something when the `mrmime` release is done.

Thanks for the heads-up! I will wait with a release then (cc: @mabiede)

**Pull request merged**